### PR TITLE
Use `is_negative` to check the sign an element of the ConversionInfo.

### PR DESCRIPTION
**Pull request by younies**

### CODE REVIEW

File: provider/datagen/src/transform/cldr/units/helpers.rs [hunk 1]
```diff
@@ -421,22 +422,18 @@ pub(in crate::provider) fn is_scientific_number(s: &str) -> bool {
     !contains_alphabetic_chars(base) && !contains_alphabetic_chars(exponent)
 }
 
-/// Transforms a fractional number into byte numerators, byte denominators, and a sign.
+/// Converts a fractional number into its byte representation for numerator and denominator, along with its sign.
```
Comment (robertbastian) [inline provider/datagen/src/transform/cldr/units/helpers.rs:425]: consider moving this to the experimental crate, something like
```rust
impl IcuRatio {
  #[cfg(feature = "datagen")]
  pub fn to_parts(&self) -> (Vec<u8>, Vec<u8>, Sign) { ... }
}
```

File: provider/datagen/src/transform/cldr/units/helpers.rs [hunk 2]
```diff
@@ -421,22 +422,18 @@ pub(in crate::provider) fn is_scientific_number(s: &str) -> bool {
...
 pub(in crate::provider) fn flatten_fraction(
     fraction: Ratio<BigInt>,
 ) -> Result<(Vec<u8>, Vec<u8>, Sign), DataError> {
```
Comment (robertbastian) [inline provider/datagen/src/transform/cldr/units/helpers.rs:428]: nit: doesn't need to return `Result`

**Review by robertbastian — APPROVED** [review]

File: provider/datagen/src/transform/cldr/units/helpers.rs (hunk 2 above)
Comment (younies) [reply provider/datagen/src/transform/cldr/units/helpers.rs:428]: good catch

File: provider/datagen/src/transform/cldr/units/helpers.rs (hunk 1 above)
Comment (younies) [reply provider/datagen/src/transform/cldr/units/helpers.rs:425]: yes,  I want to have a struct that can be converted from and to IcuRatio to be implemented in Datagen

**Pull request merged**